packages/components/textarea/src/textarea.spec.ts

```diff
@@ -0,0 +1,351 @@
+    });
+
+    it('should not have a maxLength by default', () => {
+      expect(el).not.to.have.attribute('maxLength');
+    });
+
+    it('should not be invalid by default', () => {
+      expect(el).not.to.have.attribute('invalid');
+    });
+
+    it('should not be valid by default', () => {
+      expect(el).not.to.have.attribute('valid');
+    });
+
+    it('should not have showValid by default', () => {
+      expect(el).not.to.have.attribute('showValid');
+    });
+
+    it('should not have a autocomplete attribute when autocomplete property is not provided', async() => {
+      el.autocomplete = "off";
+      await el.updateComplete;
+
+      el.autocomplete = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('autocomplete')).to.be.null;
+    });
+
+    it('should have a resize attribute of vertical when resize property is not provided', async() => {
+      el.resize = "auto";
+      await el.updateComplete;
+
+      (el.resize as ResizeType | undefined) = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('resize')).to.equal('vertical');
+    });
+
+    it('should have a wrap attribute of soft value when wrap property is not provided', async() => {
+      el.wrap = "hard";
+      await el.updateComplete;
+
+      (el.wrap as WrapType | undefined) = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('wrap')).to.equal('soft');
+    });
+
+    it('should not have maxlength attribute when maxLength property is not provided', async() => {
+      el.maxLength = 8;
+      await el.updateComplete;
+
+      el.maxLength = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('maxlength')).to.be.null;
+    });
+
+    it('should not have minlength attribute when minLength property is not provided', async() => {
+      el.minLength = 2;
+      await el.updateComplete;
+
+      el.minLength = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('minLength')).to.be.null;
+    });
+
+
+    it('should not have a placeholder attribute when placeholder property is not provided', async() => {
+      el.placeholder = "my placeholder";
+      await el.updateComplete;
+
+      el.placeholder = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('placeholder')).to.be.null;
+    });
+
+    it('should not have readonly attribute when readonly property is not provided', async() => {
+      el.readonly = true;
+      await el.updateComplete;
+
+      el.readonly = undefined;
+      await el.updateComplete;
+
+      expect(el.textarea.getAttribute('readonly')).to.be.null;
+    });
+  });
+
+  describe('resize of textarea', () => {
+    beforeEach(async () => {
+      el = await fixture(html`<sl-textarea></sl-textarea>`);
+    });
+
+    it('should set the textarea height to 100px if resize is auto and 100px height is set', async() => {
```

Comment:
```suggestion
    it('should set the textarea to the value of the scrollHeight', async() => {
```
you don't actually test it will (in the real world) be 100px. The 100px is just a random value, it shouldn't be in the name of the test